src/typescript/sdk/src/indexer-v2/queries/base.ts

```diff
@@ -0,0 +1,112 @@
+import "server-only";
+
+import { type Account } from "@aptos-labs/ts-sdk";
+import { type Period } from "../../const";
+import { ORDER_BY } from "../../queries";
+import { type AnyNumberString } from "../../types";
+import { TableName } from "../types/snake-case-types";
+import { postgrest } from "./client";
+import { toAccountAddressString } from "../../utils";
+
+const selectSwapsByMarketID = ({ marketID }: { marketID: AnyNumberString }) =>
+  postgrest
+    .from(TableName.SwapEvents)
+    .select("*")
+    .eq("market_id", marketID)
+    .order("market_nonce", ORDER_BY.DESC);
+
+const selectChatsByMarketID = ({ marketID }: { marketID: AnyNumberString }) =>
+  postgrest
+    .from(TableName.ChatEvents)
+    .select("*")
+    .eq("market_id", marketID)
+    .order("market_nonce", ORDER_BY.DESC);
+
+const selectPeriodicEventsByPeriod = ({
+  marketID,
+  period,
+  startTime,
+}: {
+  marketID: AnyNumberString;
+  period: Period;
+  startTime: Date;
+}) =>
+  postgrest
+    .from(TableName.PeriodicStateEvents)
+    .select("*")
+    .eq("market_id", marketID)
+    .eq("period", period)
+    .gte("start_time", startTime)
+    .order("start_time", ORDER_BY.ASC);
+
+const selectMarketLatestStateEvents = ({ marketID }: { marketID?: AnyNumberString }) =>
+  (typeof marketID === "undefined"
+    ? postgrest.from(TableName.MarketLatestStateEvent).select("*")
+    : postgrest.from(TableName.MarketLatestStateEvent).select("*").eq("market_id", marketID)
+  ).order("market_nonce", ORDER_BY.DESC);
```

Comment:
There should ideally be an index for this.

I guess this (the order by) is to get the latest updated markets right ?

I think it will mostly be ok, but if we get over 10K markets we might run into slow queries without an index.

(This is a note, feel free to mark as resolved once read.)

Reply:
Discussed on call, this will be in another PR- the index will now be on market nonce and not bump time!